Support basic Markdown formatting

nbgoodall opened the pull request:
This PR adds basic support for Slack's markdown formatting, specifically **bold**, ~~strikethrough~~ and [links](https://example.com). Non-supported features like headings and tables are left as-is.

Closes #249.

dblock commented on lib/slack/messages/formatting.rb:
```diff
@@ -68,6 +68,21 @@ def user_link(user_id)
+
+          # convert links
+          slack_md.gsub(/\[(.*?)\]\((.*?)\)/, '<\2|\1>')
+        end
```
Let’s chain them as `text.gsub…gsub…` to avoid the temporary variable.

dblock reviewed (changes requested):
Question: are we going to run into situations where the order of these replacements will matter? Like `**bold_italic_**bold` - meaning I wonder whether it’s worth implementing something that actually reads tokens sequentially.

Either way I’m good with a first cut like this. Nitpicks follow.

nbgoodall commented:
> Question: are we going to run into situations where the order of these replacements will matter? Like `**bold_italic_**bold` - meaning I wonder whether it’s worth implementing something that actually reads tokens sequentially.

@dblock I won't say never, although I haven't found any glaring cases. It wouldn't have converted `*italic*`, which I've now fixed (also accounting for `***bold and italic***` to `*_bold and italic_*`). If other issues come up I'll happily address them, and I've added another spec for nested formatting.

dblock commented:
Merged, thank you!
